threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,84 @@
+
+![Notation verifies artifacts stored in registries using verification plugin](../media/notation-verify-plugin.svg)
+
+The following diagram illustrates Notation verifying artifacts stored as OCI image layout in the file system:
+
+![Notation verifies artifacts as OCI image layout in the filesystem](../media/notation-verify-local.svg)
+
+The certificates trusted by the verifier are stored in Notation trust store in the file system. Users can download the certificates from KMS or request CA vendor for CA certificates.
+
+### Threats and Mitigation
+
+|   Title                      | STRIDE threat type     | Threat Status | Priority | Entity          | Description   | Mitigation   |
+| ---------------------------- | ---------------------- | ------------- | -------- | --------------- | ------------- | ------------ |
+| Inaccessible Registry        | Denial of Service      | Mitigated     | High     | Registry        | Registry is not able to service incoming requests or perform up to spec, thus users are unable to verify artifacts | Users verify signatures stored locally |
+| Serving tampered artifacts   | Tampering              | Mitigated     | High     | Registry        | Data stored in registries are tampered, this may lead to malicious artifacts being consumed | Users verify artifacts before consuming |
+| Disclosure of signature info | Information disclosure | Mitigated     | High     | Registry        | Attackers eavesdrop on registries, this may lead to the disclosure of sensitive signature info | Users should interact with trusted registries via TLS or Users can sign artifacts stored in local file system and store signature locally |
+| Tampered files               | Tampering              | Mitigated     | High     | File System      | Files read by Notation are tampered, this may lead to failure of notation verify operation or verification bypassed | Use trusted infrastructure with right file permissions |
+| Disclosure of files          | Information disclosure | Mitigated     | High     | File System      | Files are accessed by attackers, this may lead to disclosure of notation config files and credential files | Use trusted infrastructure with right file permissions. Users can use credential store to securely store the credentials |
+| Malicious plugin             | Tampering              | Mitigated     | High     | Notation Plugin | A malicious plugin is installed, this may lead to arbitrary code being executed by attackers | Verify plugin package before installation or plugin is installed by authorized users |
+| Inaccessible OCSP Responder  | Denial of Service      | Mitigated     | High     | OCSP Responder  | OCSP Responder is not able to service incoming requests or perform up to spec, thus users are unable to validate certificate revocation status | Users can configure trust policy to skip certificate revocation status check |
```

Comment:
Another open discussion here, for `Inaccessible OCSP Responder` the Mitigation doesn't sound very solid. Skipping certificate revocation status check means trusting the signing key NOT been compromised during `Sign`, which may not be true. As a mitigation for threat model, perhaps `switch to Revocation Checking with CRL` is a better option?

Reply:
Revocation checking with CRL was not supported by Notation v1 release, so your proposed mitigation doesn't work. Any idea from @shizhMSFT and @priteshbandi

Reply:
If OCSP Responder inaccessible then Notation cannot mitigate the threat (I would say its unmitigated) because it’s not in notation purview to control OCSP behaviour and verification should fail if revocation is configured as enforce and OCSP is inaccessible.
The best user can do is to change revocation level to warn to bypass the issue but that again is not a mitigation.

Reply:
You may consider OSCP is a more efficient replacement of CRL but they all require network access.

I agree with @priteshbandi that we move revocation check out of the critical path to avoid availability issues.